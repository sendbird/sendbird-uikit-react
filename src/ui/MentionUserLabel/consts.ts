export const MENTION_USER_LABEL_CLASSNAME = 'sendbird-mention-user-label';
