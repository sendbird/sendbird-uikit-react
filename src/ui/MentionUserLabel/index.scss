@import '../../styles/variables';

.sendbird-mention-user-label {
  &:hover {
    cursor: pointer;
  }
  display: inline-block;
  font-family: var(--sendbird-font-family-default);
  font-size: 14px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.43;
  letter-spacing: normal;
  white-space: pre-line;
  width: fit-content;
  max-width: 100%;
  height: 16px;
  @include themed() {
    color: t(on-bg-1);
  }
  &.reverse {
    @include themed() {
      color: t(on-content-1);
    }
  }
  &.purple {
    font-family: Roboto;
    font-size: 18px;
    font-weight: bold;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: normal;
    @include themed() {
      color: t(primary-3);
    }
  }
}
