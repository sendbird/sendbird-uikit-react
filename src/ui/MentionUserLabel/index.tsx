import React from 'react';
import './index.scss';

interface MentionUserLabelProps {
  className?: string
  children?: string;
  isReverse?: boolean;
  color?: string;
  userId?: string;
}

export default function MentionUserLabel({
  className = '',
  children,
  isReverse = false,
  color,
  userId,
}: MentionUserLabelProps): JSX.Element {
  return (
    <span
      className={`sendbird-mention-user-label ${className} ${isReverse ? 'reverse' : ''} ${color}`}
      contentEditable={false}
      data-userid={userId}
    >
      {children}
    </span>
  );
}
