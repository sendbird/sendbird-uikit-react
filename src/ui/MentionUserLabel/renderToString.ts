// cretes a sanitized string from a mention user label
import DOMPurify from 'dompurify';

type renderToStringParams = {
  userId: string;
  nickname: string;
};

export default function renderToString({ userId, nickname }: renderToStringParams): string {
  // donot change this template, it wont work
  const el = `<span data-userid="${userId}" data-sb-mention="true" class="sendbird-mention-user-label">${nickname}</span>`;
  const purifier = DOMPurify(window);
  const sanitized_ = purifier.sanitize(el);
  const token = sanitized_.split(' ');
  const [spanTag, ...rest] = token;
  // we do this because DOMPurify removes the contenteditable attribute
  const sanitized = [spanTag, 'contenteditable="false"', ...rest].join(' ');
  return sanitized;
}
