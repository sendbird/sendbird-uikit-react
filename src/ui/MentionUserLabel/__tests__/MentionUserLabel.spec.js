import React from 'react';
import { render } from '@testing-library/react';

import MentionUserLabel from "../index";

describe('ui/MentionUserLabel', () => {
  it('should do a snapshot test of the MentionUserLabel DOM', function() {
    const text = "example-text";
    const { asFragment } = render(<MentionUserLabel />);
    expect(asFragment()).toMatchSnapshot();
  });
});
