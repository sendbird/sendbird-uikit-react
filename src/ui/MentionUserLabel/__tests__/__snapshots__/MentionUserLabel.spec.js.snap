// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentionUserLabel should do a snapshot test of the MentionUserLabel DOM 1`] = `
<span
  className="sendbird-mention-user-label   undefined"
  contentEditable={false}
/>
`;
